src/artemis/external_interaction/nexus/write_nexus.py

```diff
@@ -285,6 +121,117 @@ def update_nexus_file_timestamp(self):
             shutil.copy(filename, temp_filename)
             with h5py.File(temp_filename, "r+") as nxsfile:
                 nxsfile["entry"].create_dataset(
-                    "end_time", data=np.string_(self._get_current_time())
+                    "end_time", data=np.string_(get_current_time())
                 )
             shutil.move(temp_filename, filename)
+
+    def get_image_datafiles(self, max_images_per_file=1000):
+        return [
+            self.directory / f"{self.filename}_{h5_num + 1:06}.h5"
+            for h5_num in range(-(-self.full_num_of_images // max_images_per_file))
+        ]
+
+
+class FGSNexusWriter(NexusWriter):
+    grid_scan: dict
+
+    def __init__(self, parameters: FGSInternalParameters, grid_scan: dict) -> None:
+        super().__init__(parameters)
+        self.goniometer = create_gridscan_goniometer_axes(
+            parameters.artemis_params.detector_params,
+            parameters.experiment_params,
+            grid_scan,
+        )
+        self.grid_scan = grid_scan
+
+    def _get_data_shape_for_vds(self) -> tuple[int | float, ...]:
+        ax = list(self.grid_scan.keys())[0]
+        num_frames_in_vds = len(self.grid_scan[ax])
+        nexus_detector_params: EigerDetector = self.detector.detector_params
+        return (num_frames_in_vds, *nexus_detector_params.image_size)
```

Comment:
I think this function might fall out as being just in the parent class but this is probably an issue for the following PR

Reply:
it is implemented properly in the following PR, where it is just `(total frames, *image_size)`